simulator/simulator.h

```diff
@@ -29,6 +29,20 @@ class Simulator : public Log {
 
     static std::unique_ptr<Simulator> create_simulator( const std::string& isa, bool functional_only, bool log);
     static std::unique_ptr<Simulator> create_configured_simulator();
+
+    virtual size_t sizeof_register() const = 0;
+    template <typename T>
+    T read_cpu_register( uint8 regno) {
+        return static_cast<T>( this->read_cpu_register_internal( regno));
+    }
```

Comment:
Please put a whitespace between methods, otherwise they are misinterpreted as parts of a single if-then or try-catch block.